docs/EXTENDED-ARCHIVES.md

```diff
@@ -0,0 +1,428 @@
+
+The way this works is that you provide a generic url that redirects to the
+latest image resource. The client will follow all redirects until it finds the
+final image, and will check (using the `etag` property) if the resource matches
+the currently selected image.
+
+If the ETags don't match, the client may point the user to download the new one
+image.
+
+If the HTTP response from `updateUrl` doesn't contain an ETag this mechanism is
+ignored.
+
+##### `etag (String)`
+
+The [HTTP Etag][etag] of the current image.
+
+Clients may use this to check if the image resolved by `updateUrl` is different
+to the one we have locally.
+
+##### `expiryDate (String)`
+
+The timestamp to determine the expiration date of the image. The date should
+conform to [ISO 8601][iso8601] standard.
+
+Clients may use this information to refuse flashing the image.
+
+#### Extensions
+
+Each of the entities described above accept an object property called
+`extensions`, which you can use to provide custom properties to the manifests,
+in a way that are specific to your application. It is encouraged that you put
+those custom properties inside the `extensions` namespace to avoid potential
+clashes with other properties that may be introduced in the future.
+
+For example:
+
+```json
+{
+  "publisher": {
+    ..
+    "extensions": {
+      ..
+    }
+    ..
+  },
+  "device": {
+    ..
+    "extensions": {
+      ..
+    }
+    ..
+  },
+  "image": {
+    ..
+    "extensions": {
+      ..
+    }
+    ..
+  }
+}
+```
+
+### `.meta/publisher.svg`
+
+An [`svg`][svg] logo of the publisher.
+
+### `.meta/device.svg`
+
+An [`svg`][svg] logo of the device.
+
+### `.meta/schema.json`
+
+A [Reconfix][reconfix] image configuration schema.
+
+### `.meta/image.bmap`
+
+A [bmap][bmap] file for the extended archive image.
+
+### `.meta/instructions.markdown`
+
+A markdown file including post-flash instructions.
+
+### `.meta/release-notes.txt`
+
+A plain text file describing the image version's release notes. The client
+should read this file only if the `releaseNotesUrl` on the `image` section of
+the `.meta/manifest.json` is not defined, returns an unsuccessful HTTP code,
+or the client is offline.
+
+### `image.<extension>`
+
+The image file **in uncompressed form**. Clients should not attempt to
+decompress the image file since the extended archive already includes
+compression. If the image is compressed, then the client should write
+compressed bytes to the drive.
+
+An archive containing more than one image file is not a well-formed archive. If
+this is the case, use only the first image.
+
+### `.meta/checksum`
+
+For example: `64c7ed611929ea5178fbb69b5a5f29cc9cc7c157`
+
+This file contains the checksum matching the type specified in the
+`checksumType` field of the image section of `.meta/manifest.json`.
+
+The reason the actual checksum is at the end of the archive is that if any tool
+modifies the image inside the extended archive, it will need to update the
+checksum.
```

Comment:
Hmm, I guess ImageMaker would also need to read checksumType from the manifest.json, so it knows what kind of checksum to append to the image (unless ImageMaker also rewrites the checksumType in the manifest too)